ocaml/parsing/ast_mapper.ml

```diff
@@ -531,10 +527,11 @@ module E = struct
     match Jane_syntax.Expression.of_ast exp with
     | Some (jexp, attrs) -> begin
         let attrs = sub.attributes sub attrs in
-        Jane_syntax_parsing.Expression.wrap_desc ~loc ~info:attrs @@
         match sub.expr_jane_syntax sub jexp with
-        | Jexp_comprehension   c -> Jane_syntax.Comprehensions.expr_of   ~loc c
-        | Jexp_immutable_array i -> Jane_syntax.Immutable_arrays.expr_of ~loc i
+          | Jexp_comprehension   c ->
+            Jane_syntax.Comprehensions.expr_of   ~loc ~attrs c
+          | Jexp_immutable_array i ->
+            Jane_syntax.Immutable_arrays.expr_of ~loc ~attrs i
```

Comment:
You'll want to rebase this as I've moved this pattern match into `Jane_syntax` proper.